train_od_ex.py

```diff
@@ -0,0 +1,288 @@
+import os
+import random
+import numpy as np
+
+import matplotlib
+import matplotlib.pyplot as plt
+import tensorflow as tf
+
+from object_detection.utils import config_util
+from object_detection.builders import model_builder
+
+from utils import load_image_into_numpy_array
+from utils import plot_detections
+
+OD_MODELS_PATH = '.'
+os.chdir(OD_MODELS_PATH)
+# For interactive plot in PyCharm
+matplotlib.use('module://backend_interagg')
+
+
+class RubberDucksTraining:
+    def __init__(self):
+        physical_devices = tf.config.list_physical_devices('GPU')
+        for device in physical_devices:
+            tf.config.experimental.set_memory_growth(device, True)
+        self.gt_boxes = []
+        self.load_dataset()
+        self.get_gt_boxes()
+        self.preprocess_data()
+        self.create_model_from_checkpoint()
+        self.finetune()
+        self.test()
+
+    def load_dataset(self):
+        # Load images and visualize
+        train_image_dir = 'models/research/object_detection/test_images/ducky/train/'
+        self.train_images_np = []
+        for i in range(1, 6):
+            image_path = os.path.join(train_image_dir, 'robertducky' + str(i) + '.jpg')
+            self.train_images_np.append(load_image_into_numpy_array(image_path))
+
+    def visualize_img(self):
+
+        plt.rcParams['axes.grid'] = False
+        plt.rcParams['xtick.labelsize'] = False
+        plt.rcParams['ytick.labelsize'] = False
+        plt.rcParams['xtick.top'] = False
+        plt.rcParams['xtick.bottom'] = False
+        plt.rcParams['ytick.left'] = False
+        plt.rcParams['ytick.right'] = False
+        plt.rcParams['figure.figsize'] = [14, 7]
+
+        for idx, train_image_np in enumerate(self.train_images_np):
+            plt.subplot(2, 3, idx+1)
+            plt.imshow(train_image_np)
+        plt.show()
+
+
+    def get_gt_boxes(self):
+        self.gt_boxes = [
```

Comment:
Potential place to put annotations for our custom dataset

Reply:
ok